### PR TITLE
feat: kleene_or, kleene_and, and restore SQL semantics

### PR DESCRIPTION
**Pull request by danking**
I noticed that our row filters disagreed with DuckDB's semantics which is SQL semantics, in particular `where true or null` _keeps_ the row (we filtered it).

cc: @robert3005 we'll probably conflict on #1272

### CODE REVIEW

File: vortex-expr/src/binary.rs [hunk 1]
```diff
@@ -50,8 +50,8 @@ impl VortexExpr for BinaryExpr {
             Operator::Lte => compare(lhs, rhs, ArrayOperator::Lte),
             Operator::Gt => compare(lhs, rhs, ArrayOperator::Gt),
             Operator::Gte => compare(lhs, rhs, ArrayOperator::Gte),
-            Operator::And => and(lhs, rhs),
-            Operator::Or => or(lhs, rhs),
+            Operator::And => and_kleene(lhs, rhs),
+            Operator::Or => or_kleene(lhs, rhs),
```
Comment (robert3005) [inline vortex-expr/src/binary.rs:54]: I don't think we want this, we should have another operator for this or just separate expression

Comment (robert3005) [reply vortex-expr/src/binary.rs:54]: HMMM, ok we do want this

File: vortex-array/src/array/constant/compute.rs [hunk 2]
```diff
@@ -119,40 +119,74 @@ impl MaybeCompareFn for ConstantArray {
     }
 }
 
+fn arrow_and(left: Option<bool>, right: Option<bool>) -> Option<bool> {
```
Comment (robert3005) [inline vortex-array/src/array/constant/compute.rs]: call this `and`?

File: vortex-array/src/array/constant/compute.rs [hunk 3]
```diff
@@ -119,40 +119,74 @@ impl MaybeCompareFn for ConstantArray {
...
+    match (left, right) {
+        (Some(false), _) => Some(false),
+        (_, Some(false)) => Some(false),
+        (None, _) => None,
+        (_, None) => None,
+        (Some(l), Some(r)) => Some(l & r),
+    }
+}
+
+fn arrow_or(left: Option<bool>, right: Option<bool>) -> Option<bool> {
```
Comment (robert3005) [inline vortex-array/src/array/constant/compute.rs]: `or`?

File: vortex-expr/src/binary.rs (hunk 1 above)
Comment (danking) [reply vortex-expr/src/binary.rs:54]: If we want SQL semantics, we want this.

File: vortex-array/src/array/constant/compute.rs (hunk 2 above)
Comment (danking) [reply vortex-array/src/array/constant/compute.rs]: done.

File: vortex-array/src/array/constant/compute.rs (hunk 3 above)
Comment (danking) [reply vortex-array/src/array/constant/compute.rs]: done.

**Review by robert3005 — APPROVED** [review]

**Comment by robert3005** [comment]
There should be a place in RowFilter that needs changing as well?

**Comment by danking** [comment]
see: https://github.com/spiraldb/vortex/pull/1289

**Pull request merged**